### DISCUSSION TITLE
Setting up my first BirdNetPi station, problems & questions

### DISCUSSION THREAD

**Discussion by PaulRB** [mcguirepr89/BirdNET-Pi#1230] [Q&A] [answered]
Hi everyone, I set up my first BirdNetPi station at home: 
* Pi 4B
* 32GB USB 3 SSD
* Ugreen USB audio adapter with 2x 3.5mm TRS sockets, attached via 1m USB extension cable.
* Sony stereo lapel mic I've had for years, no model number, 3.5mm TRS plug.
* Official Pi 4 USB-C power supply

Questions:
1. I suspect I have what is a high level of mains hum. Can I do anything to reduce it? It does not reduce much if I move the Pi away from the mic. If I unplug the mic from the USB sound adapter, the mains hum disappears completely. 
2. There are a lot of wood pigeons and stock doves in my area. I can hear them in the recordings, yet they are not recognised. Could this be due to the high level of mains hum? (See attached image. I have circled the wood pigeon calls in white and magpie calls in yellow. BirdNet identifies only the magpies.
3. My Pi 4B is only 1GB RAM memory. What's the recommended memory size, and what problems occur if a lower size Pi is used? 

I will have more questions later, but that will do to begin with. Thanks for any help or advice!

Paul

![IMG_20240803_072921019](https://github.com/user-attachments/assets/701aa519-2cc2-45e1-bc48-e297bd59ccdc)
![IMG_20240803_072825173](https://github.com/user-attachments/assets/307d8b61-a170-4a95-abbe-f38c68f2f18a)
![Screenshot_20240804-202844-815](https://github.com/user-attachments/assets/b5135d6b-c997-4dc6-bad6-16a309ae32d9)

**Comment by Inachis** [comment]
Power supply? Move it as far away as possible; try another; power from battery if you have one?

Or maybe another nearby power supply, e.g. laptop - run it on battery.

Kelvin

**Comment by PaulRB** [comment]
Hi Kelvin,

Thanks for the suggestion. You were absolutely correct. 

I tried another official RPi USB-C power supply, same level of mains hum with that. 

Tried moving the PSU as far as cable allowed from the Pi. No change to level of mains hum. 

Tried powering Pi from laptop running on its battery. Hum completely gone! Only a low level of hiss remained, so the culprit is clear.

So how do other BirdNetPi users overcome this problem? Many of them must also use the official RPi USB-C power supply.

**Reply by tphakala** [reply]
I use DIY microphone with balanced connection to XLR audio interface, balanced connection eliminates RF noise like AC hum.

In your case make sure microphone cable is as far as possible from any AC leads and PSUs, if this still does not eliminate AC hum then you need another microphone with better shielding.

**Reply by PaulRB** [reply]
Thanks for that advice. 

How do I identify microphones with better shielding?

I guess all the inexpensive mics on eBay, Amazon etc are unlikely to have it. 

I also tried other USB-C power supplies I have around the house:
* Motorola 30W smartphone charger: same mains hum as with official RPi 15W PSU.
* Dell 130W charger for laptop: no hum! Just a faint background hiss, similar to powering the pi from laptop with battery power. So it is possible to manufacture a PSU that does not cause the hum.

**Reply by Inachis** [reply]
Hi Paul,
I'm using official RPi USB-C power supplies for both my BirdNET-Pi and BattyBirdNET-Pi - they're sat alongside each other - day/night flipping dosn't work when you want to ID the owls too, also nocturnal migration maybe in the future.

At least you know the source, maybe it's a bad sample. I know switched power supplies can generate more noise; I did think about adding some ferrite beads but there's mixed views on mains hum elimination.

For comparison here's a couple of screen grabs - I get many pigeons too, mostly feral.
![Screenshot 2024-08-05 at 15-24-08 birdnetpi1](https://github.com/user-attachments/assets/de6f06b8-830c-4665-9d4c-210311d66822)
![Screenshot 2024-08-05 at 15-13-45 birdnetpi1](https://github.com/user-attachments/assets/0c3a09f0-e809-4e7a-ad74-906dcbf63600)

Kelvin

**Reply by PaulRB** [reply]
Thanks again Kelvin. I don't think it's a bad sample problem because I get the same level of hum from 3 different power supplies (2x official RPi and a Motorola USB-C). The Dell laptop power supply is the only one that doesn't cause mains hum for some reason. 

So if others are using the RPi USB-C power supply without issues, maybe @tphakala is correct and it's the unshielded mic & cable that are the problem.

But if that's true, I would expect moving power supply, Pi, cables and mic, to reduce the cables picking up the hum, would have at least some effect, but it seems not. 

So I'm thinking that the hum is not being picked up by the mic or cable, it's already in the mic cable, coming from the power supply, throughout the Pi. So shielded cables won't help at all. So I have to eliminate it at source.

But that brings me back to wondering why other BirdNetPi users don't have a problem with the RPi USB-C power supply...

**Reply by tphakala** [reply]
I have RPI 3, 4 and 5 boards with RPI power supplies, none of them have any AC hum with my audio gear (Focusrite Scarlett audio interface and AOM 5024 DIY microphone with balanced connection).

**Reply by Inachis** [reply]
My BirdNET-Pi input was all purchased from amazon uk in a hurry(to replace a USB PS3 camera/microphone used to test initial setup):
- AGPTEK Lavalier Microphone, Clip-on Lapel Omnidirectional Condenser Mic with Wind Muff Type C 3.5mm Jack for Phone PC Laptop DSLR Camera Recording Interview, Podcast, Voice Dictation
- UGREEN USB External Stereo Sound Card Aux USB Audio Adapter 3.5mm Headphone Jack to USB Converter TRRS Mic-Supported Sound Adaptor Compatible with Headset Speaker PC Laptop MacBook Raspberry Pi
- rhinocables USB 2.0 Extension Cable A to A Extender Black Lead Male Plug to Female Socket for Extending Connection to Peripherals, Printers, Cameras, Mice, Keyboards (1, 3m) 

For testing another setup;
- BOYA BY-LM40 USB Lavalier Microphone Omnidirectional Condenser Lavalier Lapel for Laptop PC Macbook Clip on Mic for Video Recording,Interview,Skype,Gaming,Remote Work Streaming,Podcasting

btw, make sure there aren't any other potential noise sources in the vicinity - wallwarts, lights, etc.

Kelvin

**Reply by PaulRB** [reply]
Having finished working for the day, I am using the Dell power supply to power the RPi. Finally, it had recognised the woodies:
![Screenshot_20240805-190754](https://github.com/user-attachments/assets/5eec944e-aab5-4a00-ada3-61f80bbeaa46)

**Reply by PaulRB** [reply]
Hi Kelvin,

Why did you choose the Boya mic instead of another AGPTECH? Just to compare which was best? To avoid buying another USB sound adapter? Which have you found better, and in what ways?

**Comment by PaulRB** [comment]
Is there a way I can reduce the hum from the RPi power supply? For example with capacitors connected to the 5V and ground pins of the Pi's GPIO header?

**Reply by PaulRB** [reply]
UPDATE: I tried 1000uF, 100uF and 0.1uF caps in various combinations connected to the 5V and ground pins of the GPIO header. No effect on mains hum from the RPi USB-C power supply.

The Dell laptop USB-C gives much better results, but it is complete overkill for a Pi and I need it for the laptop anyway!

**Comment by PaulRB** [comment]
@tphakala maybe your much more expensive Focusrite Scarlett audio interface is filtering the hum from the power supply so it doesn't reach the mic.?

**Comment by PaulRB** [comment]
@Inachis Hi Kelvin,
* AGPTEK Lavalier Microphone: this seems to have a USB-C adapter. Have you tried using that instead of the UGREEN? You would need a USB-C to USB-A adapter to do that.
* UGREEN USB External Stereo Sound Card: I didn't consider that one because it has a TRRS socket and the mic, which I already had, has a TRS plug. The price is reasonable, maybe I'll pick one up, along with your AGPTEK mic.
* rhinocables USB 2.0 Extension Cable A to A: I didn't find the extension cable gave any advantage that I could perceive. It didn't do anything to reduce the mains hum, even when I used it to maximise the distance between the Pi and the mic. Now, using the Dell power supply, I've removed the USB extension cable and this hasn't caused any problem that I can hear.
* other potential noise sources: using the Dell power supply, there is no mains hum, so there can't be any other noise sources around that are causing it. The official RPi power supply does cause it, I've tried 2 of them, but one of your other components must be blocking or filtering it out for you, I think. Question is, is it the Pi itself, the USB audio adapter or the mic....?

@Inachis @tphakala thanks for your help so far, it's turning out to be an interesting troubleshooting exercise!

**Comment by Inachis** [comment]
Hi Paul,
-  AGPTEK Lavalier Microphone: with USB-C adapter - not tried
- I needed the extension for convenient setup - the microphone is under the soffit routing the cable through a bedroom window, ditto for the AudioMoth(configured as USB microphone) to the BattyBirdNET-Pi.

I was in hurry but looking back I probably followed https://github.com/mcguirepr89/BirdNET-Pi/discussions/39#discussioncomment-1490164 for the AGPTEK and UGREEN recommendation? Interesting to note that thread also mentions hum...

Kelvin
PS During the early days of BattyBirdNET-Pi I discovered with my bat detector that the RPi USB-C power supply creates a lot of ultrasonic noise so they are as far as possible from the AudioMoth.

**Comment by PaulRB** [comment]
I'm continuing to experiment and learn with what I have available. 

I quickly discovered that the Dell laptop power supply, although it dramatically reduces the hum noise, isn't suitable for the Pi. It is meant to be a laptop charger, and it's capacity is 130W which is 10~20 times what the Pi needs. It shuts off after about 1 hour use with the Pi. If I unplug and re-plug it from the Pi, it runs for another hour and shuts off. I think this behaviour is due to it being designed to charge a laptop's battery and is a power-saving measure ment to cut in after the laptop's battery is fully charged.

So I'm back to using the official RPi power supply. But something I did while fiddling around with it has caused the level of hum to reduce significantly. The hum is not gone, but much better than before. More species being identified now. I need to play around some more to try to figure out what it was I did that changed things!

**Reply by Inachis** [reply]
If you haven't already tried, waggle cable and connections, might be a loose joint or badly soldered connection?

Kelvin

**Reply by PaulRB** [reply]
Will try that later, because the loud hum is back again....

**Reply by PaulRB** [reply]
Waggling any of the cables made no noticeable difference to the hum. 

I have a theory about why the hum improved temporarily. The mic may have got a little damp in some light rain. It's not supposed to be an outdoor/weatherproof mic, and I didn't put in in a particularly well-sheltered spot. I noticed some droplets on it, wiped them off, and I think it was after that that I noticed the reduced hum. Maybe the hum returned as it dried out.

Still looking for another mic to buy and try. I have both the models you mentioned in Amazon basket. As always, Amazon reviews are contradictory and the use cases of the reviewers don't really match BirdNetPi use case.

**Reply by PaulRB** [reply, marked as answer]
Still haven't purchased a mic, but...

At the bottom of a box of old junk, found an analog headset with separate 3.5mm mic and headphone plugs. Plugged in to the UGREEN and found there was very little hum at all (hard to hear for sure over today's high winds).

So I think the Sony mic I was using before was probably faulty, which was not causing the hum, that was clearly the power supply, but wasn't filtering out or blocking the hum from interfering with the audio signal in some way.

**Reply by PaulRB** [reply]
Took a look inside the Sony mic. Couldn't see anything wrong... Wonder what those diodes are for, must look that up.

![IMG_20240807_194405876](https://github.com/user-attachments/assets/06210b1e-c30e-478e-b91e-cc86bdf0dcda)
